lib/codegen/fromcto/java/javavisitor.js

```diff
@@ -278,7 +278,12 @@ class JavaVisitor {
                 parameters.fileWriter.writeLine(1, '}');
                 break;
             default:
-                parameters.fileWriter.writeLine(1, 'private ' + fieldType + ' ' + fieldName + ';' );
+                if (this.isMap(field)) {
+                    const decl = field.getModelFile().getAllDeclarations().find(d => d.name === field.ast.type.name);
+                    parameters.fileWriter.writeLine(1, `private Map<${this.toJavaType(decl.getKey().getType())}, ${this.toJavaType(decl.getValue().getType())}> ${field.getName()} = new HashMap<>();`);
```

Comment:
Just double checking, can date objects be hashed in Java?

Reply:
@ekarademir Yes `java.util.Date` can be hashed